### ISSUE TITLE
Verify ref_mols format for v2 cset/RHS upload

### ISSUE THREAD

**Issue by mwinokan** [m2ms/fragalysis-frontend#1327]
@mwinokan to verify if the observation shorthand e.g. x0152a works, and assign dev as necessary

**Comment by mwinokan** [comment]
I have been testing the RHS upload on v2 staging with target A71EV2A and the following SDF: [test_upload_1.sdf.zip](https://github.com/m2ms/fragalysis-frontend/files/14196103/test_upload_1.sdf.zip) 

I have tried:

* using ref_mols in the format: `ax0473a` which is exactly as it is copied from the LHS UI.
* using ref_mols in the format: `x0473a` which is what we want to be the actual value

@kaliif can you please change the backend to accept observations in the format `x0473a` and change the string copied from the LHS UI from `ax0473a` to `x0473a` unless there is a reason for the prepended **a**

**Comment by phraenquex** [comment]
@mwinokan no, if it's ax0473a after upload, then that's the final code, that everyone should use.

**Comment by Waztom** [comment]
@kaliif please check if RHS upload is using the long LHS name vs the shortened LHS dataset name.

**Comment by kaliif** [comment]
@mwinokan what is this field?
>  <ref_pdb>  (1) 
x0310_with_Z3621038789.pdb

**Comment by mwinokan** [comment]
That is the file name of the reference protein PDB (the upload doesn't support the PDB's yet)

**Comment by kaliif** [comment]
How is this constructed? ZINC id's are a recent additions, what did this field look like before? Asking because it's currently blowing up the upload.

**Comment by mwinokan** [comment]
@kaliif According to #1249 the `rhs_pdb` as far as I know should be any string referring to a PDB file in the uploaded ZIP archive. In previous test SDF's 
[XX01ZVNS2B_manual_v2_fix-2 (1).sdf.zip](https://github.com/m2ms/fragalysis-frontend/files/14211026/XX01ZVNS2B_manual_v2_fix-2.1.sdf.zip) it has been a short string like `x0051_0B`. For the SDF `test_upload_1.sdf` I generated some virtual hits of the x0310 crystal with a ligand, hence the ZINC ID.

**Comment by phraenquex** [comment]
@mwinokan: Kalev's question was about `<ref_pdfb>` - is that field even valid anymore?   (`rhs_pdb` is for yellow release, remember.)

**Comment by kaliif** [comment]
How do you generate those files? This target upload doesn't match the version I have, so I don't have anything resembling ax0310a in the database ( not x0310a, x0310* or even x03*). I'm not sure how much I can edit this file by hand and it still should be uploadable. Is it only those 2 fields that determine the match or is there something else?

**Comment by kaliif** [comment]
I've made some fixes to the uploader. This probably still doesn't work but to continue I need some further input. @mwinokan, @Waztom, can you prepare a new upload file?

**Comment by mwinokan** [comment]
Sure @kaliif I will prepare one. Is the A71EV2A target ok? And are there any new columns you need example data for?

**Comment by kaliif** [comment]
A71EV2A is fine. I don't know about the columns, what's the  minimum set it should accept. The previous file you gave me failed because of the lack of columns, but we're not sure if this is by design or a code bug.

**Comment by kaliif** [comment]
What happens when you try to upload it to stack?

**Comment by Waztom** [comment]
@kaliif can you please try this upload file to test the RHS (Have used the same LHS observation short codes in Staging today). I get this, which might be expected, now that we are dealing with multiple obervations. 

The upload file I used is => 

[MW_2A_12Feb2024_Fragmenstein_WT_update.txt](https://github.com/m2ms/fragalysis-frontend/files/14241344/MW_2A_12Feb2024_Fragmenstein_WT_update.txt)

![image](https://github.com/m2ms/fragalysis-frontend/assets/24798762/a96e7ea0-e1af-41d0-8918-a99a42809a10)

The problem is that multiple

**Comment by mwinokan** [comment]
@kaliif I tried another upload with a modified version of Warren's file with the following changes:

[MW_2A_12Feb2024_Fragmenstein_WT_update_fix.sdf.zip](https://github.com/m2ms/fragalysis-frontend/files/14250753/MW_2A_12Feb2024_Fragmenstein_WT_update_fix.sdf.zip)

* I added fields to the header molecule `ref_mols`, `ref_pdb`, or `original SMILES` which are present in the other non-header molecules. (Probably not causing an error, but just best practice)
* I prepended the observations with the letter 'A', i.e. `ax0375a` and now I get errors such as:

> No SiteObservation code contains 'ax0473a'

So the frontend needs to be updated to show the correct named observations, e.g. `ax0473a`
and the backend needs to compare the `ref_mols` and `ref_pdb` values against this form of the observation name as well.

**Comment by Waztom** [comment]
@mwinokan you will get errors because the prepended `a` name is not specified in #1277 and not used in the current LHS short codes. I am less concerned about prepending vs. being able to differentiate (both Fragalysis RHS to LHS and humans talking about the data) between different canonical sites, for example (see the pic above plus [comment](https://github.com/m2ms/fragalysis-frontend/issues/1277#issuecomment-1939751650) in #1277.

**Comment by mwinokan** [comment]
@kaliif Thank you for your hard work, I was able to upload the following SDF for the A71EV2A target!

[filtered_2A_iter2_ROCS_fragalysis.sdf.zip](https://github.com/m2ms/fragalysis-frontend/files/14308696/filtered_2A_iter2_ROCS_fragalysis.sdf.zip)

[Staging Snapshot](https://fragalysis.xchem.diamond.ac.uk/viewer/react/projects/22/22)

The ref_mols column is behaving as expected, I was able to reference LHS observations using their shortcodes (from copying and consistent with downloaded structures). There are of course other bugs but I will spin those into a new ticket.

**Comment by mwinokan** [comment]
@kaliif the only thing that I was not able to achieve was to assign a reference LHS observation for the RHS [P] and [S] buttons. I added a column in the SDF called `ref_pdb` but this did not work.

**Comment by mwinokan** [comment]
> the only thing that I was not able to achieve was to assign a reference LHS observation for the RHS [P] and [S] buttons. I added a column in the SDF called ref_pdb but this did not work.

@boriskovar-m2ms maybe the above issue is actually frontend?

**Comment by Waztom** [comment]
@mwinokan the [P] and [S] buttons (via [F] toggle) are linked to the <ref_mols>. The [P] and [S] buttons for the actual follow-up RHS compounds (I see are greyed out) is linked to the <ref_pdb> that can either be a custom .pdb uploaded in the .zip or a LHS observation. Looks like you have done everything as it should be in your .sdf file.

@boriskovar-m2ms I can not see any API error codes when I load the RHS compounds - @kaliif or actually @alanbchristie can you gleam anything from the logs? The error is related to a .pdb file not being associated with the RHS dataset - this feels like an error in the RHS upload code (I suspect the regex is the reason for this). 

![image](https://github.com/m2ms/fragalysis-frontend/assets/24798762/940f5fe0-ea3e-42c0-87fa-880d0baf379e)

**Comment by mwinokan** [comment]
@boriskovar-m2ms see the SDF here: https://github.com/m2ms/fragalysis-frontend/issues/1327#issuecomment-1947994765

**Comment by kaliif** [comment]
@Waztom when I'm trying to upload, I'm getting a lot of warnings like this:
```
2024-02-16T11:12:37+0000 viewer.cset_upload.get_site_observation():180 WARNING # Molecule <rdkit.Chem.rdchem.Mol object at 0x7f382d559460> has no "lhs_pdb" property (A71EV2A, A71-ROCS-iter2-2024-02-16-A A71EV2A)
2024-02-16T11:12:37+0000 viewer.cset_upload.set_mol():376 WARNING # Failed to get a LHS SiteObservation (lhs_pdb) for <rdkit.Chem.rdchem.Mol object at 0x7f382d559460>, A71EV2A, A71-ROCS-iter2-2024-02-16-A A71EV2A
```

IS `lhs_pdb` property maybe missing in the upload file?

**Comment by Waztom** [comment]
@kaliif I see @alanbchristie added the <lhs_pdb> option to the upload code - see [here](https://github.com/xchem/fragalysis-backend/commit/a3aa0c43052e2d579232a3e0268637eea023eeea). I tried replacing <ref_pdb> with <lhs_pdb> and get an upload fail. 

![image](https://github.com/m2ms/fragalysis-frontend/assets/24798762/ccf2d672-dd3b-4f2c-811d-3841d0b89f57)

@alanbchristie this addition is not included in the RHS current V1.2 upload [spec](https://github.com/xchem/fragalysis-backend/blob/7016c1b09052301ca5a4c64d8f07447913597309/design_docs/computed_set_upload.md#L1) and also seems to be causing issues between the validator and upload function - possibly also not tested?

Previously <ref_pdb> covered both custom uploaded .pdbs and/or linking with the LHS observations. Any reason not to revert to only using the <ref_Pdb>?

**Comment by Waztom** [comment]
@kaliif confirms that the RHS data is being served via the similar/same endpoints to V1. @boriskovar-m2ms to check what needs to change in the frontend - specifically the amount of work required.

**Comment by phraenquex** [comment]
@mwinokan is this now working?  (Can't tell from the ticket - apologies if I missed the earlier discussion)
